jobs/ads-incrementality-dap-collector/ads_incrementality_dap_collector/main.py

```diff
@@ -0,0 +1,77 @@
+import click
+import logging
+import traceback
+
+from constants import LOG_FILE_NAME
+from helpers import (
+    get_config,
+    get_experiment,
+    prepare_results_rows,
+    collect_dap_results,
+    write_job_logs_to_bucket,
+    write_results_to_bq,
+)
+
+
+@click.command()
+@click.option("--gcp_project", help="GCP project id", required=True)
+@click.option(
+    "--job_config_bucket",
+    help="GCS bucket where the configuration for this job can be found. See example_config.json for format details.",
+    required=True,
+)
+@click.option(
+    "--auth_token",
+    envvar="DAP_AUTH_TOKEN",
+    help="The 'token' defined in the collector credentials, used to authenticate to the leader",
+    required=True,
+)
+@click.option(
+    "--hpke_private_key",
+    envvar="DAP_PRIVATE_KEY",
+    help="The 'private_key' defined in the collector credentials, used to decrypt shares from the leader and helper",
+    required=True,
+)
+def main(gcp_project, job_config_bucket, auth_token, hpke_private_key):
+    try:
+        logging.info(
+            f"Starting collector job with configuration from gcs bucket: {job_config_bucket}"
+        )
+        config = get_config(
+            gcp_project, job_config_bucket, auth_token, hpke_private_key
+        )
+        logging.info(
+            f"Starting collector job for experiments: {config.nimbus.experiments}."
+        )
+
+        for experiment_config in config.nimbus.experiments:
+            try:
+                experiment = get_experiment(experiment_config, config.nimbus.api_url)
+
+                tasks_to_collect = prepare_results_rows(experiment)
+
+                collected_tasks = collect_dap_results(
+                    tasks_to_collect, config.dap, experiment_config
+                )
+
+                write_results_to_bq(collected_tasks, config.bq)
+            except Exception as e:
+                logging.error(
+                    f"Collector job failed for {experiment_config.slug}. Error: {e}\n{traceback.format_exc()}"
```

Comment:
Change exception flow so that we throw from this job to airflow if any exception happens (and don't try to keep looping over the experiments).